core/signal.c

```diff
@@ -467,8 +467,7 @@ int uwsgi_signal_wait(struct wsgi_request *wsgi_req, int signum) {
 	return received_signal;
 }
 
-void uwsgi_receive_signal(struct wsgi_request *wsgi_req, int fd, char *name, int id) {
-
+pid_t uwsgi_receive_signal(struct wsgi_request *wsgi_req, int fd, char *name, int id) {
```

Comment:
the spooler call `uwsgi_receive_signal` is at `file: core/spooler.c, line: 468`.

when a signal is routed to spooler,  maybe all of the spooler process will be wakeup,
and call the `uwsgi_receive_signal`.
But only one spooler process will success at `ssize_t ret = read(fd, &uwsgi_signal, 1);`
in `file: core/signal.c, line: 474`

So, `uwsgi_receive_signal` should return a sign to indicate whether this spooler process will run the signal handler.


I think this can be simply return 0 for success read ( will run signal handler ), -1 for nothing.